test/corpus/old_global_variables.txt

```diff
@@ -0,0 +1,372 @@
+            (comment)
+            (string_literal))
+        )
+    (old_global_variable_declaration
+        (variable_storage_class)
+        (old_variable_declaration
+            (symbol)
+            (dimension)
+            (array_litteral
+                (comment)
+                (int_literal)
+                (int_literal)
+                (int_literal)
+                (int_literal)
+                (int_literal)
+                (int_literal)
+                (int_literal)
+                (int_literal)
+                (int_literal)
+                (int_literal)
+                (comment)
+                (int_literal)
+                (int_literal)
+                (int_literal)
+                (int_literal)
+                (int_literal)
+                (int_literal)
+                (comment)))
+        )
+    (old_global_variable_declaration
+        (variable_storage_class)
+        (old_variable_declaration
+            (symbol)
+            (dimension)
+            (array_litteral
+                (binary_expression
+                    (int_literal)
+                    (int_literal))
+                (comment)
+                (binary_expression
+                (binary_expression
+                (int_literal)
+                (int_literal))
+            (int_literal))
+            (comment)
+            (binary_expression
+                (binary_expression
+                    (int_literal)
+                    (int_literal))
+                (binary_expression
+                    (int_literal)
+                    (int_literal)))
+            (comment)
+            (binary_expression
+                (binary_expression
+                    (int_literal)
+                    (int_literal))
+                (binary_expression
+                    (int_literal)
+                    (int_literal)))
+            (comment)))
+        )
+    (old_global_variable_declaration
+        (variable_visibility)
+        (old_variable_declaration
+          (symbol) (int_literal))
+        ))
+
+
+========================
+old - optional semicolon
+========================
+
+new a
+new bool:c = true
+decl String:arg1[32]
+new Handle:h = INVALID_HANDLE
+a
+
+---
+
+(source_file
+    (old_global_variable_declaration
+        (old_variable_declaration name: (symbol)))
+    (old_global_variable_declaration
+        (old_variable_declaration
+            type: (old_type (old_builtin_type))
+            name: (symbol)
+            initialValue: (bool_literal)))
+    (old_global_variable_declaration
+        (old_variable_declaration
+            type: (old_type (old_builtin_type))
+            name: (symbol)
+            (fixed_dimension (int_literal))))
+    (old_global_variable_declaration
+        (old_variable_declaration
+            type: (old_type (symbol))
+            name: (symbol)
+            initialValue: (symbol)))
+    (old_global_variable_declaration
+        (old_variable_declaration
+            name: (symbol))))
```

Comment:
Here fails the last test for me because of the missing `;`  
I don't have a solution, because writing the grammar rules for old and new syntax is really hard :disappointed:

I would be okay, leaving this test failing for now :slightly_smiling_face:

Reply:
This is acceptable IMO 👍